src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,50 @@
-import React from 'react';
+import { TodoCompletedCategory } from '../../types/todoCompletedCategory';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
+
+const nameForCategoriesOption = {
+  [TodoCompletedCategory.all]: 'All',
+  [TodoCompletedCategory.active]: 'Active',
+  [TodoCompletedCategory.completed]: 'Completed',
+};
+
+export const TodoFilter = () => {
+  const { query, category: todoCategory } = useAppSelector(
+    state => state.filter,
+  );
+  const dispatch = useAppDispatch();
+
+  const changeQuery = (newQuery: string) => {
+    dispatch(filterSlice.actions.changeQuery(newQuery));
+  };
+
+  const changeTodoCategory = (newCategory: TodoCompletedCategory) => {
+    dispatch(filterSlice.actions.changeStatus(newCategory));
+  };
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={todoCategory}
+            onChange={event => {
+              const selectedCategory = event.target
+                .value as TodoCompletedCategory;
```

Comment:
The type casting here assumes that the value will always be a valid `TodoCompletedCategory`. Ensure that the `value` is indeed a valid category before casting, or handle potential errors if the casting fails.